cmd/api/src/api/middleware/middleware.go

```
@@ -331,3 +333,96 @@ func EnsureRequestBodyClosed() mux.MiddlewareFunc {
+}
+
+// MetricsMiddleware wires the API request pipeline to the Prometheus metrics declared in metrics.go. It must be
+// registered as pre-route middleware so that unmatched requests are still counted; the muxRouter argument is used to
+// resolve the matched route template for the "handler" label without dispatching the request twice.
+func MetricsMiddleware(muxRouter *mux.Router) mux.MiddlewareFunc {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			handlerLabel := routeTemplateFor(muxRouter, r)
+			curriedDuration, err := ApiRequestDuration.CurryWith(prometheus.Labels{"handler": normalizeHandlerLabel(handlerLabel)})
+			if err != nil {
+				slog.ErrorContext(r.Context(), "failed to curry request duration metric", attr.Error(err))
```

golangci-lint (2.12.1)

[major] message should be capitalized

(sloglint)